### DISCUSSION TITLE
Can this be used to make a converter of HID-Gamepad to HID-Keyboard?

### DISCUSSION THREAD

**Discussion by eduo** [NicoHood/HID#348] [Q&A] [answered]
Hi

I just discovered this and I think this could work for what I want, but wanted to confirm since it would be my first Arduino project if so.

A hobby of mine and my children is building el cheapo joysticks from kits widely available. These are later used to play arcade or console emulators, which in turn makes them interested in the history of videogames and helps them be interested in programming and DIY.

While sharing their hobby with friends they sometimes face situations where the end device doesn't support HID-Gamepads but it does support HID-Keyboard profiles.

So I thought perhaps I could use something like this to build a "translator" that can take up to 4 HID-Gamepad USB inputs and translate to HID-Keyboard presses (predefined to a layout like [this](https://user-images.githubusercontent.com/154630/156397229-3ccf5723-37f1-4372-b650-d4ea7ae151bf.png).

This way these joysticks could be used to work in other devices (like iPads, which support mame and emulatores but not HID-Gamepad).

**Comment by NicoHood** [comment, marked as answer]
You'd need an usb host shield and read the gamepad and write as a keyboard. I dont know how complex this is, but if the usb host shield can recognize it, it SHOULD be fairly easy. For PC there are easy software solutions.

**Reply by eduo** [reply]
Thank you. I need it to run on hardware as asking to install software may be a bit violent for other people's machines :D
